tensorflow_recommenders_addons/dynamic_embedding/python/ops/hkv_hashtable_ops.py

```diff
@@ -0,0 +1,517 @@
+          table's key_dtype.
+        dynamic_default_values: The values to use if a key is missing in the
+          table. If None (by default), the static default_value
+          `self._default_value` will be used.
+        return_exists: if True, will return a additional Tensor which indicates
+          if or not keys are existing in the table.
+        name: A name for the operation (optional).
+
+      Returns:
+        A tensor containing the values in the same shape as `keys` using the
+          table's value type.
+        exists:
+          A bool type Tensor of the same shape as `keys` which indicates
+            if keys are existing in the table.
+            Only provided if `return_exists` is True.
+
+      Raises:
+        TypeError: when `keys` do not match the table data types.
+    """
+    with ops.name_scope(
+        name,
+        "%s_lookup_table_find" % self.name,
+        (self.resource_handle, keys, self._default_value),
+    ):
+      keys = ops.convert_to_tensor(keys, dtype=self._key_dtype, name="keys")
+      with ops.colocate_with(self.resource_handle, ignore_existing=True):
+        if return_exists:
+          values, exists = hkv_ops.tfra_hkv_hash_table_find_with_exists(
+              self.resource_handle,
+              keys,
+              dynamic_default_values
+              if dynamic_default_values is not None else self._default_value,
+          )
+        else:
+          values = hkv_ops.tfra_hkv_hash_table_find(
+              self.resource_handle,
+              keys,
+              dynamic_default_values
+              if dynamic_default_values is not None else self._default_value,
+          )
+    return (values, exists) if return_exists else values
+
+  def insert(self, keys, values, name=None, shared_name=None):
+    """Associates `keys` with `values`.
+
+        Args:
+          keys: Keys to insert. Can be a tensor of any shape. Must match the table's
+            key type.
+          values: Values to be associated with keys. Must be a tensor of the same
+            shape as `keys` and match the table's value type.
+          name: A name for the operation (optional).
+
+        Returns:
+          The created Operation.
+
+        Raises:
+          TypeError: when `keys` or `values` doesn't match the table data
+            types.
+        """
+    with ops.name_scope(
+        name,
+        "%s_lookup_table_insert" % self.name,
+        [self.resource_handle, keys, values],
+    ):
+      keys = ops.convert_to_tensor(keys, self._key_dtype, name="keys")
+      values = ops.convert_to_tensor(values, self._value_dtype, name="values")
+      with ops.colocate_with(self.resource_handle, ignore_existing=True):
+        # pylint: disable=protected-access
+        op = hkv_ops.tfra_hkv_hash_table_insert(self.resource_handle, keys,
+                                                values)
+    return op
+
+  def accum(self, keys, values_or_deltas, exists, name=None, shared_name=None):
```

Comment:
Could you add `scores=None` after the `exists` for reserving the customized strategy supporting?